yew-macro/src/html_tree/html_component.rs

```diff
@@ -414,6 +414,13 @@ impl Parse for Props {
                 return Err(syn::Error::new_spanned(&prop.label, "expected identifier"));
             }
 
+            if prop.question_mark.is_some() {
+                return Err(syn::Error::new_spanned(
+                    &prop.label,
+                    "optional attributes are only supported on HTML tags. Yew components can use `Option<T>` properties to accomplish the same thing.",
+                ));
+            }
+
```

Comment:
That reminds me, why I'm not in flavor of `?` syntax. Thats create unnecessary difference between built-in and custom components with no good reason

Reply:
That difference is in no way unnecessary. All attributes are technically optional for normal html elements but the same cannot be said for components where the component type controls which props are optional and which are required.

You could argue that the syntax should work for optional Component props but that's a different issue.

Reply:
@siku2 could we avoid using `?` if we did `Into<Option<...>>`? Or does it run into issues?

Reply:
> @siku2 could we avoid using `?` if we did `Into<Option<...>>`? Or does it run into issues?

@jstarry
I talked about that here: https://github.com/yewstack/yew/issues/903#issuecomment-680808862.
The second point no longer stands because we unified the runtime interface.

Do you feel it would be more ergonomic to drop the explicit syntax in favor of `Into<Option<_>>`?

Reply:
> Alternatively, we can also adjust the behaviour of fields marked with `prop_or*` so that they take `Into<Option<T>>` values where `None` acts the same way as if the prop wasn't given.

Yes, that is exactly what I prefer to see

Reply:
I opened a new issue for this: #1550